File: backend_pet/src/main/java/com/example/backend_pet/entity/Product.java
package com.example.backend_pet.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product extends BaseEntity {

    @NotBlank
    @Column(nullable = false)
    private String name;

    @NotNull
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(name = "image_url")
    private String imageUrl;

    @NotBlank
    @Column(nullable = false)
    private String category;

    @NotNull
    @PositiveOrZero
    @Column(nullable = false)
    @Builder.Default
    private Integer stock = 0;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String brand;

    private String weight;

    private String volume;

    private String material;
}

File: fontend_pet/src/types/product.ts
// Product types
export interface Product {
  id: number;
  name: string;
  price: number;
  imageUrl: string;
  category: string; // Mã danh mục (slug): "food", "accessories", "grooming"
  stock: number;
  description?: string;
  brand?: string;
  weight?: string;
  volume?: string;
  material?: string;
}
